python/dlr/dlr_model.py

```diff
@@ -140,20 +112,69 @@ def __init__(self, model_path, dev_type='cpu', dev_id=0):
     def __del__(self):
         if getattr(self, "handle", None) is not None and self.handle is not None:
             if getattr(self, "lib", None) is not None:
-                _check_call(_LIB.DeleteDLRModel(byref(self.handle)))
+                self._check_call(DLRModelImpl._LIB.DeleteDLRModel(byref(self.handle)))
             self.handle = None
 
+    def _lazy_init_output_shape(self):
+        self.output_shapes = []
+        self.output_size_dim = []
+        for i in range(self.num_outputs):
+            shape = self._get_output_shape(i)
+            self.output_shapes.append(shape)
+
+    def _parse_backend(self):
+        backend = c_char_p()
+        self._check_call(DLRModelImpl._LIB.GetDLRBackend(byref(self.handle),
+                                       byref(backend)))
+        return backend.value.decode('ascii')
+
+    def _get_version(self):
+        version = c_char_p()
+        self._check_call(DLRModelImpl._LIB.GetDLRVersion(byref(version)))
+        return version.value.decode('ascii')
+
+    def _check_call(self, ret):
+        """
+        Check the return value of C API call
+        This function will raise exception when error occurs.
+        Wrap every API call with this function
+
+        Parameters
+        ----------
+        ret : int
+            return value from API calls
+        """
+        if ret != 0:
+            raise DLRError(DLRModelImpl._LIB.DLRGetLastError().decode('ascii'))
+
+    def _init_libdlr(self):
+        if DLRModelImpl._LIB is None:
```

Comment:
if DLRModelImpl._LIB is not None:
    log.info("already loaded") and return
the rest of the code can be shifted to the left